fields/evaluate.go

```diff
@@ -21,113 +21,190 @@ import (
-	AggregateCount
-	AggregateMin
-	AggregateMax
-	AggregateSum
-	AggregateAvg
-	AggregateStateHistSeconds
-	AggregateStateHistPercent
-	AggregateStateHistRate
+	TimeAggregationDefault TimeAggregationMethod = iota
+	TimeAggregationCount
+	TimeAggregationMin
+	TimeAggregationMax
+	TimeAggregationSum
+	TimeAggregationAvg
+	TimeAggregationSeconds
+	TimeAggregationPercent
+	TimeAggregationRate
 )
 
-type AggregateMethod uint8
+type TimeAggregationMethod uint8
+
+const (
+	GroupAggregationDefault GroupAggregation = iota
+	GroupAggregationCount
+	GroupAggregationMin
+	GroupAggregationMax
+	GroupAggregationSum
+	GroupAggregationAvg
+)
+
+type GroupAggregation uint8
 
 var (
-	_ encoding.TextMarshaler   = AggregateMethod(0)
-	_ encoding.TextUnmarshaler = (*AggregateMethod)(nil)
+	_ encoding.TextMarshaler   = TimeAggregationMethod(0)
+	_ encoding.TextUnmarshaler = (*TimeAggregationMethod)(nil)
 )
 
-func (m AggregateMethod) String() string {
+func (m TimeAggregationMethod) String() string {
 	b, err := m.MarshalText()
 	if err != nil {
 		return "%(INVALID)!"
 	}
 	return string(b)
 }
 
-func (m AggregateMethod) MarshalText() ([]byte, error) {
+func (m TimeAggregationMethod) MarshalText() ([]byte, error) {
 	switch m {
-	case AggregateDefault:
+	case TimeAggregationDefault:
 		return nil, nil
-	case AggregateCount:
+	case TimeAggregationCount:
 		return []byte("count"), nil
-	case AggregateMin:
+	case TimeAggregationMin:
 		return []byte("min"), nil
-	case AggregateMax:
+	case TimeAggregationMax:
 		return []byte("max"), nil
-	case AggregateSum:
+	case TimeAggregationSum:
 		return []byte("sum"), nil
-	case AggregateAvg:
+	case TimeAggregationAvg:
 		return []byte("avg"), nil
-	case AggregateStateHistSeconds:
+	case TimeAggregationSeconds:
 		return []byte("state-histogram-seconds"), nil
-	case AggregateStateHistPercent:
+	case TimeAggregationPercent:
 		return []byte("state-histogram-percent"), nil
-	case AggregateStateHistRate:
+	case TimeAggregationRate:
 		return []byte("state-histogram-rate"), nil
 	}
-	return nil, fmt.Errorf("unknown aggregation method")
+	return nil, fmt.Errorf("bad aggregation method")
 }
 
-func (m *AggregateMethod) UnmarshalText(data []byte) error {
+func (m *TimeAggregationMethod) UnmarshalText(data []byte) error {
 	switch string(data) {
 	case "":
-		*m = AggregateDefault
+		*m = TimeAggregationDefault
 	case "count":
-		*m = AggregateCount
+		*m = TimeAggregationCount
 	case "min":
-		*m = AggregateMin
+		*m = TimeAggregationMin
 	case "max":
-		*m = AggregateMax
+		*m = TimeAggregationMax
 	case "sum":
-		*m = AggregateSum
+		*m = TimeAggregationSum
 	case "avg":
-		*m = AggregateAvg
+		*m = TimeAggregationAvg
 	case "state-histogram-seconds":
-		*m = AggregateStateHistSeconds
+		*m = TimeAggregationSeconds
 	case "state-histogram-percent":
-		*m = AggregateStateHistPercent
+		*m = TimeAggregationPercent
 	case "state-histogram-rate":
-		*m = AggregateStateHistRate
+		*m = TimeAggregationRate
```

Comment:
Update here as well:

Use:
```
case "state-seconds", "state-histogram-seconds":
```
To handle both the old and the new aggregation names